tests/basic/pure_test.go

```diff
@@ -4132,3 +4132,111 @@ var _ = Describe("{CreateNewPoolsWhenFadaFbdaVolumeCreationInProgress}", func()
 	})
 
 })
+
+var _ = Describe("{CreateCloneOfTheFADAVolume}", func() {
+	/*
+				https://purestorage.atlassian.net/browse/PTX-24002
+			    1.Deploy a FADA app
+		        2.Take the corresponding pxctl volume of the pvc and try to clone it
+		        3.Check the corresponding volume clone is available in FA backend
+		        4.Delete the FADA app
+
+	*/
+	JustBeforeEach(func() {
+		log.Infof("Starting Torpedo tests ")
+		StartTorpedoTest("CreateCloneOfTheFADAVolume",
+			"Create Clone of the FADA Volume and verify the status",
+			nil, 0)
+	})
+	itLog := "Px Volume Resize in parallel to FADA/FBDA Volume Resize ( PVC Resize )"
+	It(itLog, func() {
+		var contexts []*scheduler.Context
+		var volumeName string
+		var cloneVolumeId string
+		flashArrays, err := GetFADetailsUsed()
+		log.FailOnError(err, "Failed to get FA details used")
+		log.InfoD(itLog)
+		stepLog := "Deploy FADA app"
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			context, err := Inst().S.Schedule("deploy-fada", scheduler.ScheduleOptions{
+				AppKeys:            []string{"fio-fa-davol"},
+				StorageProvisioner: fmt.Sprintf("%v", portworx.PortworxCsi),
+				PvcSize:            6 * units.GiB,
+			})
+			log.FailOnError(err, "Failed to schedule application of %v namespace", "deploy-fada-volume")
+			contexts = append(contexts, context...)
+
+		})
+		ValidateApplications(contexts)
+
+		stepLog = "Get the corresponding Px volume for the PVC and clone the volume "
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			for _, context := range contexts {
+				appsvols, err := Inst().S.GetVolumes(context)
+				log.FailOnError(err, "Failed to get volumes for the context")
+				for _, vol := range appsvols[:1] {
+					cloneVolumeId, err := Inst().V.CloneVolume(vol.ID)
+					log.FailOnError(err, "Failed to clone volume [%v]", vol.ID)
+					log.InfoD("Clone Volume ID [%v]", cloneVolumeId)
+
+				}
+			}
+			log.InfoD("Get the corresponding volume name for the volId")
+			for _, context := range contexts {
+				appsvols, err := Inst().S.GetVolumes(context)
+				log.FailOnError(err, "Failed to get volumes for the context")
+				for _, vol := range appsvols {
+					if vol.ID == cloneVolumeId {
+						volumeName = vol.Name
+					}
+				}
+			}
+		})
+		checkVolumeExistsInFlashArrays := func(volumeName string, flashArrays []pureutils.FlashArrayEntry) error {
+			cloneVolFound := false
+			for _, fa := range flashArrays {
+				faClient, err := pureutils.PureCreateClientAndConnect(fa.MgmtEndPoint, fa.APIToken)
+				if err != nil {
+					log.Errorf("Failed to connect to FA using Mgmt IP [%v]", fa.MgmtEndPoint)
+					continue
```

Comment:
Better to assert here instead of continue if you are not able to connect to Flash Array

Reply:
yes , faClient will nil if err and we will have nil panic in the next line

Reply:
added